lib/extension.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+// The name of the package in order to give the Antora logger a useful name
+const { name: packageName } = require('../package.json')
+const generateIndex = require('./generate-index')
+
+function register (
+  pipeline,
+  { config: { indexLatestOnly = false, languages = ['en'], ...extraArguments } }
+) {
+  const logger = pipeline.require('@antora/logger').get(packageName)
+
+  if (Object.keys(extraArguments).length > 0) {
+    const args = Object.keys(extraArguments)
+      .map((x) => `"${x}"`)
+      .join(', ')
+    logger.fatal(`Remove unrecognised extension option(s): ${args}`)
+    pipeline.halt()
+  }
```

Comment:
I'm not sure about that, could you please explain what is the purpose?

Reply:
@Mogztter The intention is to prevent spelling mistakes or entirely invalid parameters to the pipeline extension from being silently ignored. It's mostly the former scenario I want to prevent, such as:
```yaml
pipeline:
  extensions:
    - require: antora-lunr
      langugaes: ['no', 'en']
```
silently passing through, with the user not understanding why the index hasn't changed. Additionally, it prevents old or deprecated configuration parameters from remaining in use after they're no longer valid.

WDYT, @Mogztter? Would you prefer a warning?

---

Per the behaviour of `pipeline.halt` and Dan's suggestions, I'd change it to this.
```suggestion
  if (Object.keys(extraArguments).length > 0) {
    const args = Object.keys(extraArguments)
      .map((x) => `"${x}"`)
      .join(', ')
    throw new Error(`Unrecognised extension option(s) for ${extensionName}: ${args}`)
  }
```

Reply:
I like the idea of validating the config options and of throwing an error if an invalid option is used. We want to make sure the user is on the right path.

Reply:
I've changed it to throw an error rather than halt the pipeline quietly.